File: cpp/131-140/Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        
        vector<int> v(32, 0);
        for(auto e : nums)
        {
            for(int i = 0;i < 32;i++)
            {
                v[i] += e & 1;
                e = e >> 1;
            }
        }
        
        int result(0);
        for(int i = 0;i < 32;i++)
            if(v[i] % 3 != 0)
                result |= (1 << i);
                
        return result;
        
    }
};

File: python/131-140/Single Number II.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def singleNumber(self, nums):
        tmp = set(nums)
        v = sum(tmp)*3 -sum(nums)
        return v/2


s = Solution()
items = [2,2,3,2]
print s.singleNumber(items)